src/passes/OptimizeInstructions.cpp

```diff
@@ -537,6 +536,11 @@ struct OptimizeInstructions
               (op = makeUnsignedBinaryOp(binary->op)) != InvalidBinary &&
               Bits::getMaxBits(binary->left, this) <= 31) {
             binary->op = op;
+          } else if (c < 0 && c > std::numeric_limits<int32_t>::min() &&
```

Comment:
Doesn't this work for `std::numeric_limits<int32_t>::min()` as well?

`X / 2^31 == 0 iff 0 <= X < 2^31`

`X / 2^31 == 1 iff 2^31 <= X < 2^32`

Edit: Oh, I see, it's because we want to use a shift in that case. It would be good to add a comment about that.

Reply:
> Oh, I see, it's because we want to use a shift in that case. It would be good to add a comment about that.

Yes, and shifts more preferable. Will add comment